src/event.rs

```diff
@@ -336,333 +204,445 @@ pub enum WindowEvent<'a> {
     /// * Moving the window to a display with a different scale factor.
     ///
     /// After this event callback has been processed, the window will be resized to whatever value
-    /// is pointed to by the `new_inner_size` reference. By default, this will contain the size suggested
+    /// is pointed to by the `PhysicalSize` reference. By default, this will contain the size suggested
     /// by the OS, but it can be changed to any value.
     ///
     /// For more information about DPI in general, see the [`dpi`](crate::dpi) module.
-    ScaleFactorChanged {
-        scale_factor: f64,
-        new_inner_size: &'a mut PhysicalSize<u32>,
-    },
+    ScaleFactorChanged(f64, &'a mut PhysicalSize<u32>),
+}
 
-    /// The system window theme has changed.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum RawPointerEvent {
+    /// A device has been added.
+    Added,
+    /// A device has been removed.
+    Removed,
+    Press(RawPointerPress),
+    /// Relative change in physical position of a pointing device.
     ///
-    /// Applications might wish to react to this to change the theme of the content of the window
-    /// when the system changes the window theme.
+    /// This represents raw, unfiltered physical motion, NOT the position of the mouse. Accordingly,
+    /// the values provided here are the change in position of the mouse since the previous
+    /// `MovedRelative` event.
+    MovedRelative(PhysicalDelta<f64>),
+    /// Change in absolute position of a pointing device.
     ///
-    /// At the moment this is only supported on Windows.
-    ThemeChanged(Theme),
+    /// The `PhysicalPosition` value is the new position of the cursor relative to the desktop. This
+    /// generally doesn't get output by standard mouse devices, but can get output from tablet devices.
+    MovedAbsolute(PhysicalPosition<f64>),
+    /// Change in rotation of mouse wheel.
+    Wheel(UnitlessDelta<f64>),
+}
+
+/// Raw keyboard events.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum RawKeyboardEvent {
+    /// A keyboard device has been added.
+    Added,
+    /// A keyboard device has been removed.
+    Removed,
```

Comment:
I'm also not sure about the usefulness of those 2 events, we're not a compositor in the end and tracking keyboards is strange. Is it related to multiseat discussion?

Reply:
In case both of them were added purely for multiseat support it's actually would be pretty useful, however I'm not sure if platforms other than Wayland has something like that under the hood. For Wayland implementing those should be trivial.

Reply:
The OS exposes keyboard connect and disconnect events as part of the device event API, so it's fairly trivial for us to do so as well. The equivalent of these events in the old API were [`DeviceEvent::Added` and `Removed`](https://docs.rs/winit/0.22.2/winit/event/enum.DeviceEvent.html#variant.Added).

Reply:
Oh, I guess I've just never seen those events comming(because Wayland backend doesn't send them now at all), and I'm not sure how to even handle keyboard connect/disconnect on Wayland, since it's not really something visible to a client. I guess it won't have them at all in the end.

Reply:
Right now, it looks like the only platforms with solid keyboard device events are Windows and X11. On those platforms, device events are tied to the physical devices that emit them and the `DeviceId`s/`{Pointer|Keyboard}DeviceId`s are unique for each physical device. The documentation should be more clear about the state of things on the other platforms, though.

Reply:
For reference, last I checked (a while ago) Wayland still hadn't nailed down what they were going to do for raw input, since the windows/X11 way of doing things is considered a privacy hazard due to the ease of implementing keyloggers. Maybe in the future there'll be an explicit authorization scheme.